plugins/services/src/js/structs/Framework.js

```diff
@@ -66,6 +123,55 @@ module.exports = class Framework extends Application {
     return this._spec;
   }
 
+  getCheckStatus() {
```

Comment:
Is this supposed to be used by consumers of the Framework struct? So far it looks like it is just used internally by `getStatus`, `getServiceStatus`. I wonder if it would be better to move to a util?

Reply:
Thinking about this again after we discussed this in standup where the concern was that we would have to pass `this` to a util function - would it make sense to move to a util but pass tasks as param so there's no dependency on `this`? I can see there being confusion about when/where to use `getCheckStatus` vs `getStatus` vs `getServiceStatus` but maybe we could resolve that with some clarifying code comments.

Reply:
i'd second that. factoring a function out of the Framework-class (not necessarily into a new file) and passing tasks would be a good thing.